core/workspace_svg.js

```diff
@@ -1702,10 +1853,11 @@ Blockly.WorkspaceSvg.prototype.setScale = function(newScale) {
   if (this.grid_) {
     this.grid_.update(this.scale);
   }
+  this.scroll_(this.scrollX, this.scrollY);
```

Comment:
We call scroll_ instead of scrollbar.resize() so that we can center the zoom correctly without scrollbars, but scroll_ does not resize the scrollbars so we have to call resizeContent as well.

Also added the above as a comment.